exercises/saddle-points/example.rs

```diff
@@ -1,23 +1,24 @@
 pub fn find_saddle_points(input: &[Vec<u64>]) -> Vec<(usize, usize)> {
-    let mut saddle_points = Vec::new();
+    let (width, height) = match (input.len(), input[0].len()) {
+        (0, _) | (_, 0) => return Vec::new(),
+        (n, k) => (n, k),
+    };
 
-    let width = input.len();
-    let height = input[0].len();
+    let mut saddle_points = Vec::new();
 
     for i in 0..width {
-        for j in 0..height {
-            let column = input.iter().map(|x| x[j]).collect::<Vec<u64>>();
-            let row = &input[i];
+        let row_max = input[i].iter().max().unwrap();
```

Comment:
```suggestion
        let row_max = *input[i].iter().max().unwrap();
```

May as well dereference the maximum here; that'll make the value comparison in line 17 prettier.

Reply:
Agree, good point.